### PR TITLE
WIP: implementation of cr16 cross-compile

### PR DESCRIPTION
**Pull request by siraben**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Ericson2314** [comment]
Oh nice, this one doesn't need a GCC or binutils forks?

**Comment by siraben** [comment]
GCC builds but cannot produce executables
```
$ nix-build -A pkgsCross.cr16.hello   
these derivations will be built:
  /nix/store/qkqz7wpmszv8l48xkcb4w2shp1x3ygq1-hello-2.10-cr16-elf.drv
building '/nix/store/qkqz7wpmszv8l48xkcb4w2shp1x3ygq1-hello-2.10-cr16-elf.drv'...
unpacking sources
unpacking source archive /nix/store/3x7dwzq014bblazs7kq20p9hyzz0qh8g-hello-2.10.tar.gz
source root is hello-2.10
setting SOURCE_DATE_EPOCH to timestamp 1416139241 of file hello-2.10/ChangeLog
patching sources
updateAutotoolsGnuConfigScriptsPhase
Updating Autotools / GNU config script to a newer upstream version: ./build-aux/config.sub
Updating Autotools / GNU config script to a newer upstream version: ./build-aux/config.guess
configuring
configure flags: --disable-dependency-tracking --prefix=/nix/store/6l4kqxw7ngsmn8zrcb9frig2m5n3ckbn-hello-2.10-cr16-elf --build=x86_64-apple-darwin --host=cr16-elf
checking for a BSD-compatible install... /nix/store/g77kammvqqn63v01ppig43m2yn2s1lqd-coreutils-8.31/bin/install -c
checking whether build environment is sane... yes
checking for cr16-elf-strip... cr16-elf-strip
checking for a thread-safe mkdir -p... /nix/store/g77kammvqqn63v01ppig43m2yn2s1lqd-coreutils-8.31/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking for cr16-elf-gcc... cr16-elf-gcc
checking whether the C compiler works... no
configure: error: in `/private/tmp/nix-build-hello-2.10-cr16-elf.drv-0/hello-2.10':
configure: error: C compiler cannot create executables
See `config.log' for more details
builder for '/nix/store/qkqz7wpmszv8l48xkcb4w2shp1x3ygq1-hello-2.10-cr16-elf.drv' failed with exit code 77
error: build of '/nix/store/qkqz7wpmszv8l48xkcb4w2shp1x3ygq1-hello-2.10-cr16-elf.drv' faile
```

**Comment by Ericson2314** [comment]
@siraben Can you add the `WIP: ` then to all your PRs which aren't ready to merge?

**Comment by siraben** [comment]
@GrahamcOfBorg build pkgs.pkgsCross.cr16.hello

**Comment by siraben** [comment]
Currently it compiles GCC successfully, however I get a linking error.

<details>
<summary>main.c</summary>

```c
int main() {
  return 0;
}
```
</details>

<details>
<summary>Log</summary>

```
$ $CC -o hello hello.c
/nix/store/dfcdwnbsyfwqc7pl0i3f4vgcw5xrrxvg-cr16-elf-binutils-2.31.1/bin/cr16-elf-ld: warning: -z now ignored
/nix/store/dfcdwnbsyfwqc7pl0i3f4vgcw5xrrxvg-cr16-elf-binutils-2.31.1/bin/cr16-elf-ld: warning: -z relro ignored
/nix/store/dfcdwnbsyfwqc7pl0i3f4vgcw5xrrxvg-cr16-elf-binutils-2.31.1/bin/cr16-elf-ld: /nix/store/hfcz93lhbsfpkpiygnj6b26ymkw62vbj-newlib-3.3.0-cr16-elf/cr16-elf/lib/crt1.o: in function `start':
(.text+0x1c): undefined reference to `_init'
/nix/store/dfcdwnbsyfwqc7pl0i3f4vgcw5xrrxvg-cr16-elf-binutils-2.31.1/bin/cr16-elf-ld: (.text+0x20): undefined reference to `_fini'
/nix/store/dfcdwnbsyfwqc7pl0i3f4vgcw5xrrxvg-cr16-elf-binutils-2.31.1/bin/cr16-elf-ld: /nix/store/hfcz93lhbsfpkpiygnj6b26ymkw62vbj-newlib-3.3.0-cr16-elf/cr16-elf/lib/libc.a(lib_a-fini.o): in function `_libc_fini_array':
/private/tmp/nix-build-newlib-3.3.0-cr16-elf.drv-0/newlib-3.3.0/cr16-elf/newlib/libc/misc/../../../.././newlib/libc/misc/fini.c:36: undefined reference to `_fini'
collect2: error: ld returned 1 exit status
```
</details>

File: pkgs/development/misc/cr16/newlib.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{ stdenv, fetchurl, texinfo, flex, bison, crossLibcStdenv, buildPackages }:
+
+let version = "2.0.0";
+in
+
+crossLibcStdenv.mkDerivation {
+  pname = "newlib";
+  inherit version;
+  src = fetchurl {
+    url = "ftp://sourceware.org/pub/newlib/newlib-${version}.tar.gz";
```
Comment (Ericson2314) [inline pkgs/development/misc/cr16/newlib.nix:10]: This is a vanilla newlib src, so it shouldn't go in in a `cr16` directory. We might also have it already packaged?

**Comment by SuperSandro2000** [comment]
Closing as stale

**Pull request closed without merging**